src/python/pants/core/target_types.py

```diff
@@ -116,29 +170,44 @@ class AssetSourceField(SingleSourceField):
 
         If an http_source is provided, represents the network location to download the source from.
         The downloaded file will exist in the sandbox in the same directory as the target.
-        `http_source` has the following signature:
-            http_source(url: str, *, len: int, sha256: str, filename: str = "")
+        `http_source`/`https_sources` have the following signatures:
+
+            http_source(url: str, *, sha256: str,  len: int,  filename: str = "")
+            http_sources(
+                *,
+                # Each of these accept a `|` separated string matching `<url>|<sha256>|<len>`
+                # where whitespace is stripped around the `|`.
+                # E.g. https://pantsbuild.org|feed6789feed6789feed6789|112233
+                linux_arm64: str | None = None,
+                linux_x86_64: str | None = None,
+                macos_arm64: str | None = None,
+                macos_x86_64: str | None = None,
+                filename: str | None = None,
+            )
+
         The filename defaults to the last part of the URL path (E.g. `example.ext`), but can also be
         specified if you wish to have control over the file name. You cannot, however, specify a
         path separator to download the file into a subdirectory (you must declare a target in desired
         subdirectory).
+
         You can easily get the len and checksum with the following command:
+
             `curl -L $URL | tee >(wc -c) >(shasum -a 256) >/dev/null`
         """
     )
 
     @classmethod
     def compute_value(  # type: ignore[override]
-        cls, raw_value: Optional[Union[str, HTTPSource]], address: Address
-    ) -> Optional[Union[str, HTTPSource]]:
+        cls, raw_value: Optional[Union[str, HTTPSource, HTTPSources]], address: Address
+    ) -> Optional[Union[str, HTTPSource, HTTPSources]]:
         if raw_value is None or isinstance(raw_value, str):
             return super().compute_value(raw_value, address)
-        elif not isinstance(raw_value, HTTPSource):
+        elif not isinstance(raw_value, (HTTPSource, HTTPSources)):
             raise InvalidFieldTypeException(
                 address,
                 cls.alias,
                 raw_value,
-                expected_type="a string or an `http_source` object",
+                expected_type="a string or an `http_source`/`http_sources` object",
```

Comment:
maybe `"a string an `http_source` or an `http_sources` object"` feels more natural..?
The "an x/y object" almost feels like you're saying it's the same thing but could be used with different names. Although, in this case maybe that is actually close enough to the truth :P ?